### PR TITLE
Refactor error input tests in test_torch.py to OpInfos

### PR DESCRIPTION
**Pull request by saketh-are**
This PR ports several tests in `test/test_torch.py` over to OpInfo ErrorInputs.

Some tests commented "convert to ErrorInputs" still remain in `test_torch.py`. They fall under two categories:
- Memory overlap tests which specifically test the in-place version of an operator (e.g. [this test](https://github.com/pytorch/pytorch/blob/424a054d53e5f3229411532fb69fae276b29dcb2/test/test_torch.py#L3788) for index_add_).
- Tests with non-trivial behavior calling `torch.cuda.synchronize()` after calling the operator being tested (e.g. [this test](https://github.com/pytorch/pytorch/blob/424a054d53e5f3229411532fb69fae276b29dcb2/test/test_torch.py#L4948) for torch.multinomial).

### CODE REVIEW

File: aten/src/ATen/native/Normalization.cpp [hunk 1]
```diff
@@ -26,7 +26,7 @@ TORCH_META_FUNC(renorm)(const Tensor& self, const Scalar& p, int64_t dim, const
   TORCH_CHECK(maxnorm.toDouble() >= 0.0,
               "renorm: expected maxnorm to be >= 0 but got ", maxnorm.toDouble());
   const auto ndim = self.dim();
-  TORCH_CHECK(ndim > 1, "renorm: input needs at least 2 dimensions, got ", ndim, "dimensions");
+  TORCH_CHECK(ndim > 1, "renorm: input needs at least 2 dimensions, got ", ndim, " dimensions");
```
Comment (mruberry) [inline aten/src/ATen/native/Normalization.cpp:29]: Nice spacing correction

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -3098,6 +3098,39 @@ def error_inputs_gather(op_info, device, **kwargs):
...
 
+    x = torch.rand((1,), device=device).expand((3,))
+    src = torch.rand((6,), device=device)
+    ind = torch.tensor([2, 1, 0], device=device, dtype=torch.int64)
+
+    yield ErrorInput(SampleInput(src, args=(0, ind,), kwargs=dict(out=x)),
+                     error_type=RuntimeError,
+                     error_regex='unsupported operation')
+
+    yield ErrorInput(SampleInput(src, args=(0, ind,), kwargs=dict(out=src)),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3109]: I suppose it's OK for error inputs to share data because they're just supposed to throw an error

**Review by mruberry — APPROVED** [review]
Cool! Thanks @saketh-are!
